contracts/Market.sol

```diff
@@ -356,7 +358,44 @@ contract Market is IMarket, Ownable, ERC721 {
 
 		emit Settled(index, _bets[index].payout, result, recipient);
 
-		_burn(index);
+		_burn(uint256(index));
+	}
+
+	/* 
+	 * @dev Reverse a bet. Return the stake to the bettor and the loan to the vault
+	 * @param index The index of the bet
+	 * @param signature Signature from market owner
+	 */
+	function refund(uint64 index, SignatureLib.Signature calldata signature) external {
+		bytes16 REFUND_CMD = 0x726566756e6400000000000000000000; //Bytes for "refund"
+		bytes32 messageHash = keccak256(
+			abi.encodePacked(REFUND_CMD, address(this), index)
+		);
+
+		require(
+			isValidSignature(messageHash, signature) == true,
+			"refund: Invalid signature"
+		);
+		_refund(index);
+	}
+
+	function _refund(uint64 index) internal virtual {
+		Bet memory bet = _bets[index];
+		require(bet.settled == false, "refund: Bet has already settled");
+
+		bet.settled = true;
+		uint256 loan = _bets[index].payout - _bets[index].amount;
+		_totalExposure -= loan;
+		_totalInPlay -= _bets[index].amount;
+		_inplayCount --;
+		
+		IERC20(_vault.asset()).transfer(ownerOf(index), bet.amount);
+		IERC20(_vault.asset()).transfer(address(_vault), loan);
+		emit Repaid(_vault.asset(), loan);
```

Comment:
afaik dont events come before state changes?

Reply:
No, events come after so you know the above line has excited.  Check the examples here: https://consensys.net/blog/developers/solidity-best-practices-for-smart-contract-security/

<img width="1100" alt="image" src="https://user-images.githubusercontent.com/8411406/231109113-5587c454-9426-4e34-9dbc-6c2d415f84d7.png">